### PR TITLE
Implemented CustomGlobalExceptionHandler and validation for dto

### PR DESCRIPTION
**Pull request by Zel1oy**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
```
Comment (irynamekh) [inline src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java:22]: I think it is worth adding at least one more method, for example, for processing an error that occurs at the service level when an element by this identifier is missing

**Review by irynamekh — CHANGES_REQUESTED** [review]
Good job)

**Review by irynamekh — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -52,6 +52,10 @@
             <artifactId>mysql-connector-j</artifactId>
             <version>8.1.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
```
Comment (nacenik) [inline pom.xml]: better to use starter

File: src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
     private String author;
+    @NotNull
+    @ISBN
     private String isbn;
+    @NotNull
+    @Min(0)
     private BigDecimal price;
+    @Length(min = 1, max = 255)
     private String description;
+    @URL
```
Comment (nacenik) [inline src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java]: spring could be not only URL. Commonly, an image is a byte array, so I'd recommend you validate this field on null and empty

File: src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
+    @NotNull
+    @Length(min = 1, max = 255)
     private String author;
+    @NotNull
+    @ISBN
     private String isbn;
+    @NotNull
+    @Min(0)
     private BigDecimal price;
+    @Length(min = 1, max = 255)
```
Comment (nacenik) [inline src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java]: Read about `Length` and `Size` annotation and choose the best one

File: src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(
+            EntityNotFoundException e
+    ) {
+        ArgumentNotValidResponse notValidResponse = new ArgumentNotValidResponse();
+        notValidResponse.setTimestamp(LocalDateTime.now());
+        notValidResponse.setStatus(HttpStatus.NOT_FOUND);
+        notValidResponse.setErrors(new String[]{e.getMessage()});
```
Comment (nacenik) [inline src/main/java/com/example/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: code duplication, you can extract to the private method and reuse it

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution, let's improve it :)

File: src/main/java/com/example/onlinebookstore/dto/ArgumentNotValidResponse.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package com.example.onlinebookstore.dto;
+
+import java.time.LocalDateTime;
+import lombok.Data;
+import org.springframework.http.HttpStatus;
+
+@Data
+public class ArgumentNotValidResponse {
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/dto/ArgumentNotValidResponse.java:8]: I think you can convert it to general exception response format and use this class for all handlers

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**